### PR TITLE
[FEAT] Pledge Factions

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description

A player can now pledge allegiance to a faction by visiting Pumpkin Plaza. Once a player pledges they will get a faction flag and they're name in the plaza will show which faction they below to. 

Screenshot 2024-04-17 at 2 17 18 PM
Screenshot 2024-04-17 at 2 17 29 PM
Screenshot 2024-04-17 at 2 17 34 PM
Screenshot 2024-04-17 at 2 17 40 PM
Screenshot 2024-04-17 at 2 17 50 PM
Screenshot 2024-04-17 at 1 00 37 PM

Fixes #issue

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Locally. Run BE and choose a faction. Check you get the banner and can place it. Check that you cannot choose another faction.

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/world/scenes/PlazaScene.ts [hunk 1]
```diff
@@ -484,101 +499,146 @@ export class PlazaScene extends BaseScene {
...
+    this.goblinsBanner = this.add
+      .image(100, 17, "goblins_banner")
+      .setDepth(1000000);
+    this.nightshadesBanner = this.add.image(90, 60, "nightshades_banner");
+    this.nightshadesBannerTop = this.add
+      .image(90, 56, "nightshades_banner_top")
+      .setDepth(1000000);
+    this.sunfloriansBanner = this.add
+      .image(35, 60, "sunflorians_banner")
+      .setDepth(1000000);
```
Comment (c-py) [inline src/features/world/scenes/PlazaScene.ts:516]:
```suggestion
    this.bumpkinsBanner = this.add
      .image(40, 17, "bumpkins_banner")
      .setDepth(17);
    this.goblinsBanner = this.add
      .image(100, 17, "goblins_banner")
      .setDepth(17);
    this.nightshadesBanner = this.add.image(90, 60, "nightshades_banner");
    this.nightshadesBannerTop = this.add
      .image(90, 56, "nightshades_banner_top")
      .setDepth(56);
    this.sunfloriansBanner = this.add
      .image(35, 60, "sunflorians_banner")
      .setDepth(60);
```

Setting depth = y may help banners display more accurately

Before:



After:

**Review by c-py — APPROVED** [review]
Awesome works flawlessly and the faction tags look epic - left one in-line suggestion on banner depths

**Comment by eliasSFL** [comment]
Hmm I'm not too sure if the `<>` is needed in the faction name, I feel that the colour with would be sufficient

**Comment by spencerdezartsmith** [comment]
> Hmm I'm not too sure if the `<>` is needed in the faction name, I feel that the colour with would be sufficient

We took inspiration from some other game. We like the look of it so will leave it for now :)

**Pull request merged**